dask/dataframe/io/parquet/arrow.py

```diff
@@ -614,25 +766,30 @@ def read_partition(
             path = piece
             row_group = None
             partition_keys = None
+            schema = None
         else:
-            # `piece` contains (path, row_group, partition_keys)
-            (path, row_group, partition_keys) = piece
+            # `piece` contains (path, row_group, partition_keys, schema)
+            (path, row_group, partition_keys, schema) = piece
 
         if not isinstance(row_group, list):
             row_group = [row_group]
 
         dfs = []
         for rg in row_group:
-            piece = pq.ParquetDatasetPiece(
-                path,
-                row_group=rg,
-                partition_keys=partition_keys,
-                open_file_func=partial(fs.open, mode="rb"),
-            )
-
-            arrow_table = cls._parquet_piece_as_arrow(
-                piece, columns, partitions, **kwargs
-            )
+            if pa_ds is not None and isinstance(rg, pa_ds.ParquetFileFragment):
+                # `rg` is already a `ParquetFileFragment`, pyarrow
+                # knows how to convert this to a `table`
+                arrow_table = rg.to_table(use_threads=False, schema=schema)
```

Comment:
I suppose the `columns` argument needs to be passed here? 
(and ideally we also pass the filter down here, to ensure data is already filtered before converting to pandas)